### PR TITLE
[SCR-39] Web-UI

### PR DESCRIPTION
**Pull request by EKCodeDevAcc**

### CODE REVIEW

File: webserver/pkgpkr/static/main.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+function triggerEvent(el, type){
+ if ('createEvent' in document) {
+      // modern browsers, IE9+
+      var e = document.createEvent('HTMLEvents');
+      e.initEvent(type, false, true);
+      el.dispatchEvent(e);
+  } else {
+      // IE 8
```
Comment (djsauble) [inline webserver/pkgpkr/static/main.js]: I think it's safe to ignore IE8. It's about 0.3% of all browsers, not worth the extra code complexity imo.

File: webserver/pkgpkr/webservice/templates/webservice/base.html [hunk 2]
```diff
@@ -1,78 +1,123 @@
...
-            {% endif %}
-            {% if request.session.github_token %}
-                <li class="nav-item">
-                    <a class="nav-link" href="{% url 'logout' %}">Logout</a>
-                </li>
-            {% endif %}
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+    <script type="text/javascript" charset="utf8" src="https://cdn.datatables.net/1.10.20/js/jquery.dataTables.js"></script>
+		<!-- Global site tag (gtag.js) - Google Analytics -->
+		<script async src="https://www.googletagmanager.com/gtag/js?id=G-3ZJTB9DHQG"></script>
```
Comment (djsauble) [inline webserver/pkgpkr/webservice/templates/webservice/base.html]: These lines don't seem to be indented properly. Are they using tabs instead of spaces?

File: webserver/pkgpkr/webservice/templates/webservice/recommendations.html [hunk 3]
```diff
@@ -1,19 +1,61 @@
...
-
+<section class="section">
+  <div class="container">
+    <div class="content is-medium">
+      <h2>RECOMMENDATIONS for `{{ repository_name }}`</h2>
+      <br>
+      <table>
+        <tr>
+          <td style="width:200px;">Selected Category: </td>
+          <td style="width:80px;"><input type="text" id="categoryName" name="categoryName" disabled style="border: none; background-color: white; font-size:20px"></td>
```
Comment (djsauble) [inline webserver/pkgpkr/webservice/templates/webservice/recommendations.html]: Inline styles are generally not best practice. Let's stick these in `webserver/pkgpkr/webservice/templates/webservice/styles.css`.

File: webserver/pkgpkr/webservice/recommender_service.py [hunk 4]
```diff
@@ -48,25 +48,37 @@ def get_recommendations(self, dependencies):
...
             if res_json.get('versions') and \
                     res_json['versions'].get(dependency_version) and \
                     res_json['versions'][dependency_version].get('keywords'):
                 d['keywords'] = res_json['versions'][dependency_version]['keywords']
 
+                # Version Date
+                dateTime = res_json['time'][dependency_version]
+
+                # Convert time format e.g. 2017-02-16T20:43:07.414Z -> 2017-02-16 20:43:07
+                date = dateTime.split('T')[0]
```
Comment (djsauble) [inline webserver/pkgpkr/webservice/recommender_service.py]: https://momentjs.com is perhaps a better way to handle date formatting. Especially since you're doing this in more than one place.

File: webserver/pkgpkr/webservice/views.py [hunk 5]
```diff
@@ -87,6 +88,14 @@ def repositories(request):
     one_or_more_with_dependencies = False
 
     for repo in repos:
+        # Updated Date
+        dateTime = repo['updatedAt']
+
+        # Convert time format e.g. 2020-03-16T13:03:34Z -> 2020-03-16 13:03:34
+        date = dateTime.split('T')[0]
```
Comment (djsauble) [inline webserver/pkgpkr/webservice/views.py]: Consider https://momentjs.com, as mentioned earlier.

File: webserver/pkgpkr/webservice/templates/webservice/base.html (hunk 2 above)
Comment (EKCodeDevAcc) [reply webserver/pkgpkr/webservice/templates/webservice/base.html]: I forgot what I used in there, I probably used tabs

File: webserver/pkgpkr/static/main.js (hunk 1 above)
Comment (EKCodeDevAcc) [reply webserver/pkgpkr/static/main.js]: Got it, deleted else part

File: webserver/pkgpkr/static/main.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+    $('#recommendationFilter').keyup( function() {
+        table.search($(this).val()).draw();
+    });
+});
+
+function categoryClick(value) {
+    var categoryName = document.getElementById('categoryName');
+    document.getElementById('categoryName').value = value;
```
[the remark this answers is not in the document]
Comment (EKCodeDevAcc) [reply webserver/pkgpkr/static/main.js]: the purpose of line 18 is display clicked value on input of recommendations.html which is line 18

File: webserver/pkgpkr/webservice/templates/webservice/base.html [hunk 7]
```diff
@@ -1,78 +1,123 @@
...
+                </a>
+                {% endif %}
+                <img src="{{ request.session.github_info.avatar_url }}" alt="Profile" style="height: 48px; width: 40px; padding: 8px 0 0 0">
+              {% endif %}
+              <div class="navbar-item">
+                {% if not request.session.github_token %}
+                <div class="buttons">
+                  <a class="button is-linkl" href="{% url 'login' %}">
+                    <i class="fas fa-sign-in-alt"></i>
+                    <strong>&nbsp;Log in</strong>
```
Comment (vpopil) [inline webserver/pkgpkr/webservice/templates/webservice/base.html]: I think all the nbsp s could be substitute with adding padding for fas class throuhgout

File: webserver/pkgpkr/webservice/templates/webservice/recommendations.html (hunk 3 above)
Comment (EKCodeDevAcc) [reply webserver/pkgpkr/webservice/templates/webservice/recommendations.html]: I see, I will try to provide id for elements instead and put its style into styles.css

File: webserver/pkgpkr/webservice/recommender_service.py (hunk 4 above)
Comment (EKCodeDevAcc) [reply webserver/pkgpkr/webservice/recommender_service.py]: So we just pass 2017-02-16T20:43:07.414Z format of time and convert it in js side?

File: webserver/pkgpkr/webservice/views.py [hunk 8]
```diff
@@ -87,6 +88,14 @@ def repositories(request):
...
+        time = dateTime.split('T')[-1][:-1]
+
+        repo['date'] = date + ' ' + time
```
Comment (vpopil) [inline webserver/pkgpkr/webservice/views.py]: you can do (for consistency):

`repo['date'] = f'{date} {time}'`

Comment (EKCodeDevAcc) [reply webserver/pkgpkr/webservice/views.py]: what do you think about moment.js that Daniel suggested above

File: webserver/pkgpkr/webservice/templates/webservice/base.html (hunk 7 above)
Comment (EKCodeDevAcc) [reply webserver/pkgpkr/webservice/templates/webservice/base.html]: Let me see if I can do that

File: webserver/pkgpkr/static/main.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (djsauble) [reply webserver/pkgpkr/static/main.js]: Right, but can't you do something like this?

```suggestion
    categoryName.value = value;
```

[the remark this answers is not in the document]
Comment (EKCodeDevAcc) [reply webserver/pkgpkr/static/main.js]: Oh... LOL
I misunderstood what you said and you are absolutely right
